pulsar/consumer_impl.go

```diff
@@ -563,6 +566,11 @@ func (c *consumer) Close() {
 		}
 		wg.Wait()
 		close(c.closeCh)
+		closed := closeChanSet[c.messageCh]
+		if !closed {
```

Comment:
closeChanSet is a global variable. The element in the set is never deleted. Would it be more reasonable to create a isClosed (bool) attribute in the `consumer` struct. It can be checked at line 569 and 570 as 
```
if !c.isClosed {
    close(c.messageCh)
}
```
I just do not understand why there is a need to use a global map to track the messageCh channel which will not be GCed. Isn't a leak?

Reply:
That's what I thought before, but in the case of `multiTopicConsumer` it causes the chan to be closed repeatedly.
https://github.com/apache/pulsar-client-go/blob/6a8e7f39aac100a285a2c190186e38b73a5c9d34/pulsar/consumer_multitopic_test.go#L46
![image](https://user-images.githubusercontent.com/15684156/179652863-b2827370-5415-4e38-a564-e656e4ee438c.png)


This is because although the consumers are multiple instances, they use the same `messageCh` and can only be closed once.

So I'd like to record the closed flag for `messageCh` via a global variable.


With regard to the `closeChanSet` leak, we can set it to `nil` in the `client.Close()` method.
https://github.com/apache/pulsar-client-go/blob/6a8e7f39aac100a285a2c190186e38b73a5c9d34/pulsar/client_impl.go#L212

![image](https://user-images.githubusercontent.com/15684156/179656514-ea84c9de-c873-4f5f-bac9-70946077262a.png)

Is there a better way?